motor/core.py

```diff
@@ -1758,11 +1758,14 @@ def __aiter__(self):
     __anext__ = next
 
     async def __aenter__(self):
+        if not self.delegate:
```

Comment:
Can we do a refactor in this class such that `_lazy_init`, `_next` and `_try_next` run their logic on an executor by default. It's rather odd that we have to manually do that part each time. For example, for lazy init this would look like:
```
    async def _lazy_init(self):
        if not self.delegate:
            loop = self.get_io_loop()
            self.delegate = await self._framework.run_on_executor(
                loop, functools.partial(
                    self._target.delegate.watch, **self._kwargs))
```

This would, in turn, change this `__aenter__` method as follows:

```
async def __aenter__(self):
    await self._lazy_init()
    return self
```

Reply:
I refactored to add a helper method which runs `run_on_executor`  instead of making the change you suggested. We need `_lazy_init` to be synchronous because it's called in the synchrotests: https://github.com/mongodb/motor/blob/8208420d613bfe1de0dfee55baf54c62b3b25564/synchro/__init__.py#L139-L140